### ISSUE TITLE
net.thucydides.core.webdriver.UnsupportedDriverException: Could not instantiate class org.openqa.selenium.firefox.FirefoxDriver

### ISSUE THREAD

**Issue by PrasadNutalapati** [serenity-bdd/serenity-core#1633]
I am using mac mojave 10.14.3, JDK 1.8, Serenity core 2.0.40 (latest) to develop my test cases. They all run perfectly well either with Chrome ver. 73(100% of time) or Firefox ver. 65.0(99% of time).
Our Jenkins is running on Redhat linux with no GUI. We installed Firefox browser( probably the latest available, I am not sure) since Chrome is not supported/ unavailable for RH Linux. This box is installed with **Mozilla Firefox version 60.5.0** When I merged code and ask my admin to run the git branch on Jenkins, I always get errors. 

> I have these in my project code itself: 
> Selenium-java :3.141.59
> Selenium-api: 3.141.59
> Selenium-server: 3.141.59
> geckodriver:0.24.0
> chromedriver:2.45.615355

Today after every test running 100% pass on my local Mac system, ran on Jenkins(RH Linux box) I am getting this error (ran 2 times).

```
net.thucydides.core.webdriver.UnsupportedDriverException: Could not instantiate class org.openqa.selenium.firefox.FirefoxDriver
	at net.thucydides.core.webdriver.WebDriverFacade.newDriverInstance(WebDriverFacade.java:154)
Caused by: net.thucydides.core.webdriver.UnsupportedDriverException: Could not instantiate new WebDriver instance of type class org.openqa.selenium.firefox.FirefoxDriver (Timed out waiting 45 seconds for Firefox to start.
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'

Caused by: org.openqa.selenium.WebDriverException: Timed out waiting 45 seconds for Firefox to start.
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'xxx', ip: 'xxx', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.0-957.5.1.el7.x86_64', java.version: '1.8.0_202'
Driver info: driver.version: FirefoxDriver

Caused by: org.openqa.selenium.WebDriverException: Timed out waiting 45 seconds for Firefox to start.
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'xxx', ip: 'xxx', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.0-957.5.1.el7.x86_64', java.version: '1.8.0_202'
Driver info: driver.version: FirefoxDriver
	at org.openqa.selenium.firefox.XpiDriverService.waitUntilAvailable(XpiDriverService.java:247)
```

**Comment by PrasadNutalapati** [comment]
According to this web site link [GeckoDriver, Selenium and Firefox Browser compatibility chart](https://firefox-source-docs.mozilla.org/testing/geckodriver/geckodriver/Support.html)
My current setup of geckodriver 0.24.0 and FF browser 60.5.0 and Selenium 3.141.59 should work fine.

But I still get UnsupportedDriverException. What gives ????

**Comment by wakaleo** [comment]
Have you asked the geckdodriver community?

**Comment by PrasadNutalapati** [comment]
Thanks John for your reply !
Yes, I posted in Mozilla forum too.
[UnsupportedDriverException](https://github.com/mozilla/geckodriver/issues/1530)
BTW, don't you see any thing on the development project side ?

**Comment by PrasadNutalapati** [comment]
BTW, there is this article about running FF in headless mode how to:
[headless testing using ff article from mozilla](https://hacks.mozilla.org/2017/12/using-headless-mode-in-firefox/)

I hope Serenity.properties file overrides all these settings.

**Comment by wakaleo** [comment]
That is indeed how the headless option works in Serenity.

**Comment by PrasadNutalapati** [comment]
### Any thing from the point of incompatibility due to version differences: ???
**On my local Mac:** 
FF: 65.0
geckodriver: 0.24.0
**On RedHat Linux Jenkins Server**
**FF: 60.5.**
geckodriver: 0.24.0

Since Serenity is same at both places, Selenium would also be the same.

**Comment by PrasadNutalapati** [comment]
****************POSSIBLY A BUG ?????*********
According to this page, [Serenity BDD Book](https://serenity-bdd.github.io/theserenitybook/latest/web-testing-in-serenity.html)
 I prepared my config.properties file as ...
```
## Chrome Browser Headless Testing
#webdriver.driver=chrome
#headless.mode = true
#drivers.windows.webdriver.chrome.driver = src/test/resources/webdriver/windows/chromedriver.exe
#drivers.mac.webdriver.chrome.driver = src/test/resources/webdriver/mac/chromedriver
#drivers.linux.webdriver.chrome.driver = src/test/resources/webdriver/linux/chromedriver

## Firefox Browser Headless Testing
webdriver.driver=firefox
gecko.firefox.options = {"args": ["-headless"]}
drivers.mac.webdriver.gecko.driver = src/test/resources/webdriver/mac/geckodriver
drivers.linux.webdriver.firefox.driver = src/test/resources/webdriver/linux/geckodriver
```
But I was having problem as reported above **UnsupportedDriverException**: 

But our Jenkins System Admin added the geckodriver to system path, then it WORKED !!!!!
Our Jenkins Server is running on RedHat Linux.

**Comment by wakaleo** [comment]
Possibly fixed in 2.0.42

**Comment by wakaleo** [comment]
Closing, presume answered
